src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,75 @@
 import React from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
 
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
-
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
-
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button
-              type="button"
-              className="delete"
-              data-cy="modal-close"
-            />
-          </header>
-
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
-
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+type Props = {
+  user: User | null,
+  todo: Todo,
+  onSetUser: React.Dispatch<React.SetStateAction<User | null>>,
+  onSelectUserId: (id: number) => void,
+  onSelectTodo: React.Dispatch<React.SetStateAction<Todo | null>>,
+};
 
-              {' by '}
+export const TodoModal: React.FC<Props> = ({
+  user,
+  todo,
+  onSetUser,
+  onSelectUserId,
+  onSelectTodo,
+}) => (
+  <div className="modal is-active" data-cy="modal">
+    <div className="modal-background" />
 
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
-              </a>
-            </p>
+    {user === null ? (
```

Comment:
I suggest to create separated state to handle is data loading

Reply:
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/80322190/7435726f-5a6d-4a8b-bbd4-309156ea4547)
Could you please explain the reason to make a separate state for this?

Reply:
States when there is data and the data is loading are completly different, though sometimes they may seem similar. You should always use `isLoading` state to track that data is loading and show spinners/skeletons. `!data` on the other hand indicates, for example, that we checked our servers and didn't find your data. Additionally you may track `hasError` state, when there is an error, Failed request. for example